tests/panther_mcp_core/test_fastmcp_integration.py

```diff
@@ -91,3 +94,64 @@ async def test_rule_alert_metrics_invalid_rule_ids():
                 },  # Invalid rule ID format with special chars
             )
         assert "Error calling tool 'get_rule_alert_metrics'" in str(exc_info.value)
+
+
+@pytest.mark.asyncio
+async def test_streaming_http_transport():
+    """Test streaming HTTP transport functionality."""
+
+    # Flag to track server status
+    server_started = threading.Event()
+    server_error = None
+
+    def run_server():
+        nonlocal server_error
+        try:
+            from mcp_panther.server import mcp
+
+            print("Starting server...")
+            mcp.run(transport="streamable-http", host="127.0.0.1", port=3001)
+        except Exception as e:
+            server_error = e
+            print(f"Server error: {e}")
+        finally:
+            server_started.set()
+
+    # Start server in background thread
+    server_thread = threading.Thread(target=run_server, daemon=True)
+    server_thread.start()
+
+    # Give server time to start
+    await asyncio.sleep(2)
+
+    # Check if server had startup errors
+    if server_error:
+        pytest.fail(f"Server failed to start: {server_error}")
+
+    try:
+        # Try basic HTTP connectivity first
+        async with httpx.AsyncClient() as http_client:
+            # Test if port is responding (404 is expected for root path)
+            try:
+                response = await http_client.get("http://127.0.0.1:3001/", timeout=5.0)
+                print(f"HTTP response status: {response.status_code}")
+                # 404 is expected - server only exposes MCP endpoint, not general web interface
+                if response.status_code not in [200, 404]:
```

Comment:
nit: maybe lock down the expected status? Actually since we get a response at all that means the server is active and we can say it worked right?